warehouse/query-core/src/test/java/datawave/query/jexl/visitors/IsNotNullPruningVisitorTest.java

```diff
@@ -0,0 +1,173 @@
+package datawave.query.jexl.visitors;
+
+import datawave.query.exceptions.InvalidQueryTreeException;
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.jexl.visitors.validate.ASTValidator;
+import org.apache.commons.jexl2.parser.ASTJexlScript;
+import org.apache.commons.jexl2.parser.ParseException;
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class IsNotNullPruningVisitorTest {
```

Comment:
Can you add a test case for the following to ensure that the visitor doesn't act in this case:
`!(FOO == null) && (FOO == 'bar' || FOO2 == 'baz')`

Reply:
Added this to the 'no op' test case and it passes just fine.

Reply:
I figured it would.  This is to make sure that if someone changes things in the future that they don't break this contract.